astroid/protocols.py

```diff
@@ -924,8 +924,7 @@ def generic_type_assigned_stmts(
     context: InferenceContext | None = None,
     assign_path: None = None,
 ) -> Generator[nodes.NodeNG, None, None]:
-    """Return empty generator (return -> raises StopIteration) so inferred value
-    is Uninferable.
+    """Hack. Return any Node so inference doesn't fail
+    when evaluating __class_getitem__. Revert if it's causing issues.
     """
-    return
-    yield
+    yield nodes.Const(None)
```

Comment:
This really isn't the "correct" solution but it works as the result is ignored anyway when inferring the `__class_getitem__` call. It just can't raise an Exception - that would lead to `Uninferable`.

The typing brain solves that by inferring the `T = TypeVar(...)` calls as new classes which is better but probably also not fully right.

_If it was actually uses, I believe we would've seen issue reports for it by now._
https://github.com/pylint-dev/astroid/blob/16da308bb3af0a13a7db88765de354355497981a/astroid/brain/brain_typing.py#L139-L144
https://github.com/pylint-dev/astroid/blob/16da308bb3af0a13a7db88765de354355497981a/astroid/brain/brain_typing.py#L47-L58